### DISCUSSION TITLE
Tuple syntax: multiple property assignment

### DISCUSSION THREAD

**Discussion by Logerfo** [dotnet/csharplang#880] [General]
I couldn't find something similar to this, hope it's not duplicate.
Would the syntax below be achievable?
```c#
public class A
{
   public int x { get; set; }
   public int y { get; set; }
}

public static class B
{
   public static void M1(this A a, int x, int y) => a.(x, y) = (x, y); //this
   public static (int x, int y) M2(this A a) => a.(x, y); //and this
}
```

**Comment by jnm2** [comment]
I much prefer `(a.x, a.y) =`.

**Comment by Logerfo** [comment]
@jnm2 it's simple when it's only one character, but imagine when you have `a.b.c.d.x` and `a.b.c.d.y`. Of course you can declare a variable, but the method won't be expression bodied. Also, in your example, I believe the getters are called twice. In my proposal, it would only be called once.

**Comment by jnm2** [comment]
I'd still prefer the statement body in that case.
With #803 you'd also be able to do this:

```cs
public static void M1(this A a, int x, int y) => a { x = x, y = y };
```

**Comment by Logerfo** [comment]
@jnm2 my proposal would also allow the following to be achieved:
```c#
public static void M1(this A a, (int x, int y) t) => a.(x, y) = t;
```
Since initializers doesn't support multiple assignments (like `a = b = c`), you would have to write the following:
```c#
public static void M1(this A a, (int x, int y) t) => a { x = t.x, y = t.y };
```
Again: it's simple when it's only one character, but imagine when `t` is not that reachable. I think both ideas are great, but one doesn't replace the other.

**Comment by MgSam** [comment]
With methods defined in the class you can already use tuple syntax to make concise assignment statements to fields/properties.

``` C#
public class A 
{
    int _x;
    int _y;
    public A(int x, int y) => (_x, _y) = (x, y);
}
```

Thanks, @MadsTorgersen for pointing this out.

If the new syntax for "extension classes" is adopted, you should be able to use this technique in that context as well.

**Comment by Logerfo** [comment]
@MgSam You're right, but that's valid only for extension methods. I used extension methods in my example to make it simple, but that's not always the case. You solution is partial and relies on a feature that's not yet implemented (and it's not a certainty that it will be), just like @jnm2's.

**Comment by jnm2** [comment]
Hey now, I haven't presented a solution because I'm not convinced there's a problem. 😉 Just bringing proposals in the same mental space to attention.

**Comment by MgSam** [comment]
@Logerfo My example is valid inside of all classes today as we speak. I was saying it is *not* currently applicable to extension methods, but that might change with the proposed "extension everything" feature.
